src/pages/VenuePage/VenuePage.tsx

```diff
@@ -115,92 +87,63 @@ const VenuePage: React.FC = () => {
 
-  // Camp and PartyMap needs to be able to modify this
-  // Currently does not work with roome
-  const location = venueName;
-  useLocationUpdateEffect(user, venueName);
+  // NOTE: User location updates
+
+  const updateUserLocationToCurrentVenue = useCallback(() => {
+    if (!userId || !venueName) return;
 
-  const newLocation = { [location]: getCurrentTimeInMilliseconds() };
-  const isNewLocation = profile?.lastSeenIn
-    ? !profile?.lastSeenIn[location]
-    : false;
+    trackLocationEntered({ userId, locationName: venueName });
+  }, [userId, venueName]);
 
-  const newLocations = {
-    ...prevLocations,
-    ...newLocation,
-  };
+  useInterval(() => {
+    updateUserLocationToCurrentVenue();
+  }, LOC_UPDATE_FREQ_MS);
 
   useEffect(() => {
-    if (
-      user &&
-      location &&
-      isNewLocation &&
-      ((!unmounted && !retainAttendance) || retainAttendance) &&
-      (!profile?.lastSeenIn || !profile?.lastSeenIn[location])
-    ) {
-      updateLocationData(
-        user,
-        location ? newLocations : prevLocations,
-        profile?.lastSeenIn
-      );
-      setUnmounted(false);
-    }
-    if (
-      user &&
-      profile &&
-      (profile.lastSeenIn === null || profile?.lastSeenIn === undefined)
-    ) {
-      updateLocationData(
-        user,
-        location ? newLocations : prevLocations,
-        profile?.lastSeenIn
-      );
-    }
-  }, [
-    isNewLocation,
-    location,
-    newLocation,
-    newLocations,
-    prevLocations,
-    profile,
-    retainAttendance,
-    unmounted,
-    user,
-  ]);
+    updateUserLocationToCurrentVenue();
+  }, [updateUserLocationToCurrentVenue]);
+
+  // useEffect(() => {
+  //   if (!userId || !venueName) return;
+  //   updateUserLocationToCurrentVenue();
+
+  // NOTE: A suggestion on how to avoid location cleaning, when two tabs were opened and one of them was closed
+
+  // document.addEventListener("visibilitychange", () => {
+  //   if (document.visibilityState === "visible") {
+  //     updateUserLocationToCurrentVenue();
+  //   }
+  // });
+  // }, [userId, venueName, updateUserLocationToCurrentVenue]);
 
   useEffect(() => {
-    const leaveRoomBeforeUnload = () => {
-      if (user && !retainAttendance) {
-        const locations = { ...prevLocations };
-        delete locations[venueName];
-        setUnmounted(true);
-        updateLocationData(user, locations, undefined);
-      }
-    };
-    window.addEventListener("beforeunload", leaveRoomBeforeUnload, false);
-    return () => {
-      window.removeEventListener("beforeunload", leaveRoomBeforeUnload, false);
-    };
-  }, [prevLocations, retainAttendance, user, venueName]);
+    if (!userId) return;
+
+    // NOTE: Clear user location on page close
+    window.addEventListener("beforeunload", () =>
+      updateLocationData(userId, {})
+    );
+  }, [userId]);
 
   useEffect(() => {
     if (
       profile?.enteredVenueIds &&
```

Comment:
We can get rid of this line and just use `profile?.enteredVenueIds?.includes(venueId)` below